designs/node-creation-removal.md

```diff
@@ -0,0 +1,369 @@
+# Karpenter Node Creation Removal
+
+*NOTE: Provisioning is used throughout this doc to reference Karpenter scheduling and spinning up a node. Note that this can apply to both the pending pod scheduling loop and also the consolidation scheduling loop. Essentially, any scheduling loop where Karpenter launches pods.*
+
+## 🔑  Summary
+
+Karpenter’s current node creation logic causes consistency issues as well as race conditions between instance and Node synchronization due to the provisioning loop’s lack of atomicity. Karpenter should remove node creation from its provisioning flow and allow the Kubelet to register the Node on the apiserver. As a result, Karpenter needs some way to model capacity that has been launched but has not yet been registered as a Node on the api-server.
+
+The recommendation is that this in-flight capacity should be modeled as a Machine CR that represents a Karpenter provisioner/consolidation scheduling decision. This has the following benefits:
+
+1. Karpenter stops creating the Node object, avoiding race conditions and conflicts with the kube-apiserver and the kubelet that are causing orphaned nodes and instances
+2. Karpenter does not need to know the node name at the end of its provisioning loop, shortening the provisioning loop for clouds that do not know the node name at instance launch time
+3. Kaprenter stops significantly mutating the Node object, avoiding causing conflicts with other Kubernetes tooling which relies on a well-known state for the Node
```

Comment:
Significantly is ambiguous here. I think we should be clear about ownerrferences and finalizers, which is another source of leakage (e.g. nodes can't be deleted if karpenter is uninstalled)

```suggestion
3. Karpenter stops significantly mutating the Node object, avoiding causing conflicts with other Kubernetes tooling which relies on a well-known state for the Node
```